src/adapter/src/coord.rs

```diff
@@ -4310,22 +4393,49 @@ impl<S: Append + 'static> Coordinator<S> {
             }
         }
         {
-            let compute = self.dataflow_client.compute(instance).unwrap();
-            for id in id_bundle.compute_ids.iter() {
-                since.extend(
-                    compute
-                        .collection(*id)
-                        .unwrap()
-                        .write_frontier
-                        .frontier()
-                        .iter()
-                        .cloned(),
-                );
+            for (instance, compute_ids) in &id_bundle.compute_ids {
+                let compute = self.dataflow_client.compute(*instance).unwrap();
+                for id in compute_ids.iter() {
+                    since.extend(
+                        compute
+                            .collection(*id)
+                            .unwrap()
+                            .write_frontier
+                            .frontier()
+                            .iter()
+                            .cloned(),
+                    );
+                }
             }
         }
         since
     }
 
+    /// The largest element not in advance of any object in the collection.
+    ///
+    /// Times that are not greater to this frontier are complete for all collections
+    /// identified as arguments.
+    fn largest_not_in_advance_of_upper(
```

Comment:
Let me know if you have a better name from this. It was extracted from `determine_timestamp`.